Library/AppEnvironment.swift

```diff
@@ -133,9 +133,21 @@ public struct AppEnvironment: AppEnvironmentType {
       ubiquitousStore: next.ubiquitousStore,
       userDefaults: next.userDefaults
     )
+
+    // If there are no more items in the stack,
+    // then the next call to AppEvironment.current will fail.
+    assert(self.stack.count > 0)
+
     return last
   }
 
+  internal static func resetStackForUnitTests() {
```

Comment:
Sadly, no way to mark this as only visible for testing, so I just made it obvious in the name.